# Faim dans le monde : +9 millions de morts par an

*Etude de santé publique FAO - Projet 3 - Nalron (septembre 2019) / ENSAE-ENSAI Formation Continue*  

---

### 7. Combien d'humains pourraient être nourris si toute la disponibilité intérieure mondiale de produits végétaux était utilisée pour de la nourriture ? 

Les résultats seront exprimés en termes de calories, de protéines, puis en pourcentage de la population mondiale.

*Consignes :*
Renseignez-vous sur la recommandation (de la FAO si possible) concernant le nombre de calories ainsi que sur le nombre de protéines par jour nécessaire à un être humain).

*Extrait de la FAO :*

Sedentary or light activity: If this PAL was from a female population, 30 to 50 years old, with mean weight of 55 kg and mean BMR of 5.40 MJ/day (1 290 kcal/day), TEE = 1.53 × 5.40 = 8.26 MJ (1 975 kcal), or 150 kJ (36 kcal)/kg/d.

Active or moderately active: If this PAL was from a female population, 20 to 25 years old, with mean weight of 57 kg and mean BMR of 5.60 MJ/day (1 338 kcal/day), TEE = 1.76 × 5.60 = 9.86 MJ (2 355 kcal), or 173 kJ (41 kcal)/kg/d.

Vigorous or vigorously active: If this PAL was from a male population, 20 to 25 years old, with mean weight of 70 kg and mean BMR of 7.30 MJ/day (1 745 kcal/day), TEE = 2.25 × 7.30 = 16.42 MJ (3 925 kcal), or 235 kJ (56 kcal)/kg/d.

*>>> **Pour les Calories** les calculs seront fait sur la base de **2 355 Kcal/jour**, soit **859575 Kcal/an.***

Chez les adultes, les recommandations FAO/OMS/UNUsont de 0,8 g/kg pour les femmes et **0,85 g/kg** pour les hommes.
[Source les bases de la nutrition FAO](http://www.fao.org/tempref/docrep/fao/004/w0073f/w0073f02.pdf)

*>>> **Pour les Protéines** les calculs seront fait sur la base de **0,85 g/kg**, poids moyen de 57 kg, soit **17684 g/an.***

#### 7.1 Résultat en termes de calories

In [29]:
# Utilisation de la 'disponibilite_interieure_kcal' précédemment trouvée
humain_nourris_veg_kcal = round((disponibilite_interieure_kcal / 859575), 2)
humain_nourris_veg_kcal

14153996479.86

*Plus de 14 Milliards de personnes pourraient être nourris*

In [30]:
df4.head()

Élément     Pays           Produit   Origine  Disponibilité intérieure Kg  \
0        Arménie  Blé               Végétale  554000000.0                   
2        Arménie  Maïs              Végétale  102000000.0                   
3        Arménie  Seigle            Végétale  1000000.0                     
4        Arménie  Avoine            Végétale  6000000.0                     
7        Arménie  Céréales, Autres  Végétale  21000000.0                    

Élément  Ratio énergie/poids (Kcal/Kg)  Ratio poids de protéines/poids total  \
0        2861.868300                    0.085297                               
2        0.000000                       0.121667                               
3        3041.666667                    0.060833                               
4        1972.972973                    0.088784                               
7        2607.142857                    0.072628                               

Élément  Disponibilité intérieure Kcal  
0        1.585475e+12                   
2        0.000000e+00                   
3        3.041667e+09                   
4        1.183784e+10                   
7        5.475000e+10

#### 7.2 Résultat en termes de protéines

In [31]:
# Utilisation du dataframe df4 sur lequel est appliquée une restriction sur le Végétal
humain_nourris_veg_kgprot = ((df4['Disponibilité intérieure Kg']* df4['Ratio poids de protéines/poids total']).sum())/17.68
humain_nourris_veg_kgprot = round(humain_nourris_veg_kgprot, 2)
humain_nourris_veg_kgprot

16515601006.81

*Plus de 16 Milliards de personnes pourraient être nourris*

#### 7.3 Résultats en pourcentage de la population mondiale

In [32]:
round(humain_nourris_veg_kcal/population_mondiale*100, 2) 

202.28

*202% de la population mondiale pourrait être nourris* 

In [33]:
round(humain_nourris_veg_kgprot/population_mondiale*100, 2)

236.03

*236% de la population mondiale pourrait être nourris* 

### 8. Combien d'humains pourraient être nourris si toute la disponibilité alimentaire en produits végétaux (Food) la nourriture végétale destinée aux animaux (Feed) et les pertes de produits végétaux (Waste) étaient utilisés pour de la nourriture ? 

Les résultats seront exprimés en termes de calories, de protéines, puis en pourcentage de la population mondiale.

In [34]:
# Création du dataframe (df8) à partir du dataframe des bilans alimentaires (df2) 
# Projection et Réduction des données pour cibler uniquement les produits végétaux
df8 = df2[df2['Origine'] == 'Végétale']

#### 8.1 Résultat en termes de calories

In [35]:
a = df8['Disponibilité alimentaire (Kcal/an)'] + df8['Aliments pour animaux'] + df8['Pertes']

humain_nourris_nourveg_kcal = (a* df8['Ratio énergie/poids (Kcal/Kg)']).sum()
humain_nourris_nourveg_kcal = round(humain_nourris_nourveg_kcal / 859575, 2)
humain_nourris_nourveg_kcal

12031710872.14

*Plus de 12 Milliards de personnes pourraient être nourris*

#### 8.2 Résultat en termes de protéines

In [36]:
humain_nourris_nourveg_kgprot = (a* df8['Ratio poids de protéines/poids total']).sum()
humain_nourris_nourveg_kgprot = round(humain_nourris_nourveg_kgprot / 17.68, 2)
humain_nourris_nourveg_kgprot

15064230447.27

*Plus de 15 Milliards de personnes pourraient être nourris*

#### 8.3 Résultats en pourcentage de la population mondiale

In [37]:
round(humain_nourris_nourveg_kcal/population_mondiale*100, 2)

171.95

*Soit 171% de la population mondiale*

In [38]:
humain_nourris_nourveg_kgprot/population_mondiale*100

215.28553117676665

*Soit 215% de la population mondiale*

### 9. Combien d'humains pourraient être nourris avec la disponibilité alimentaire mondiale ? 

Les résultats seront exprimés en termes de calories, de protéines, puis en pourcentage de la population mondiale.

In [39]:
# Création du dataframe (df9) selon la disponibilité alimentaire mondiale en Kcal et Kg de protéines
df9 = df1[['Disponibilité alimentaire (Kcal/an)', 'Disponibilité de protéines en quantité (Kg/an)']]
df9.head()

Élément  Disponibilité alimentaire (Kcal/an)  \
0        1.112684e+09                          
1        0.000000e+00                          
2        0.000000e+00                          
3        1.086605e+06                          
4        2.173210e+06                          

Élément  Disponibilité de protéines en quantité (Kg/an)  
0        33163.18460                                     
1        0.00000                                         
2        10.86605                                        
3        21.73210                                        
4        97.79445

#### 9.1 Résultat en termes de calories

In [40]:
humain_nourris_dispoalim_kcal = round(((df9['Disponibilité alimentaire (Kcal/an)']).sum() / 859575), 2)
humain_nourris_dispoalim_kcal

8566888.78

*Plus de 8,5 Millions de personnes pourraient être nourris*

#### 9.2 Résultat en termes de protéines

In [41]:
# Ma disponibilité de protéines est exprimée en Kg, elle sera *1000 pour avoir une base en grammes
humain_nourris_dispoalim_kgprot = round(((df9['Disponibilité de protéines en quantité (Kg/an)']).sum() / 17.68), 2)
humain_nourris_dispoalim_kgprot

11722804.3

*Plus de 11,7 Millions de personnes pourraient être nourris*

#### 9.3 Résultats en pourcentage de la population mondiale

In [42]:
pourcentage_humain_nourris_dispoalim_cal = round((humain_nourris_dispoalim_kcal*1000 / population_mondiale * 100),2)
pourcentage_humain_nourris_dispoalim_cal

122.43

*Soit 122% de la population mondiale*

In [43]:
pourcentage_humain_nourris_dispoalim_gprot = round((humain_nourris_dispoalim_kgprot*1000 / population_mondiale * 100),2)
pourcentage_humain_nourris_dispoalim_gprot

167.53

*Soit 167% de la population mondiale*

###  10. Quelle proportion de la population mondiale est considérée comme étant en sous-nutrition ?
A partir des données téléchargées qui concernent la sous-nutrition.

In [44]:
# Création du dataframe (df_sousnut) à partir du fichier 'sousnutrition.csv' extrait du site FAO
df_sousnut = pd.read_csv('p3_data/sousnutrition.csv')
df_sousnut.head()

Code Domaine                             Domaine  Code zone         Zone  \
0  FS           Données de la sécurité alimentaire  2          Afghanistan   
1  FS           Données de la sécurité alimentaire  2          Afghanistan   
2  FS           Données de la sécurité alimentaire  2          Afghanistan   
3  FS           Données de la sécurité alimentaire  2          Afghanistan   
4  FS           Données de la sécurité alimentaire  2          Afghanistan   

   Code Élément Élément  Code Produit  \
0  6132          Valeur  210011         
1  6132          Valeur  210011         
2  6132          Valeur  210011         
3  6132          Valeur  210011         
4  6132          Valeur  210011         

                                                              Produit  \
0  Nombre de personnes sous-alimentées (millions) (moyenne sur 3 ans)   
1  Nombre de personnes sous-alimentées (millions) (moyenne sur 3 ans)   
2  Nombre de personnes sous-alimentées (millions) (moyenne sur 3 ans)   
3  Nombre de personnes sous-alimentées (millions) (moyenne sur 3 ans)   
4  Nombre de personnes sous-alimentées (millions) (moyenne sur 3 ans)   

   Code année      Année     Unité Valeur Symbole Description du Symbole  Note  
0  20122014    2012-2014  millions  7.9    F       Estimation FAO        NaN    
1  20132015    2013-2015  millions  8.8    F       Estimation FAO        NaN    
2  20142016    2014-2016  millions  9.6    F       Estimation FAO        NaN    
3  20152017    2015-2017  millions  10.2   F       Estimation FAO        NaN    
4  20162018    2016-2018  millions  10.6   F       Estimation FAO        NaN

#### 10.1 Estimation de la population mondiale étant en sous-nutrition en 2013

In [45]:
# Les valeurs de population sont des strings, je fais un changement vers des floats
# Les valeurs <0.1 seront considérées comme nulles dans nos calculs
df_sousnut['Valeur'] = df_sousnut['Valeur'].replace('<0.1', 0)
df_sousnut['Valeur'] = df_sousnut['Valeur'].astype('float64')

In [46]:
# Recherche de la Population considérée comme étant en sous-nutrition sur la Période 2012-2014
# Les valeurs sont exprimées en Millions de personnes
pop_sousnut13 = df_sousnut['Valeur'][df_sousnut['Année'] == '2012-2014'].sum()
pop_sousnut13 = round(pop_sousnut13 *1e6, 2)
pop_sousnut13

881400000.0

*881,40 Millions de personnes dans le monde sont en sous-nutrition sur en 2013*

#### 10.2 Estimation de la population mondiale étant en sous-nutrition en 2017

In [47]:
# Pour info sur la Période 2016-2018 nous sommes à :
pop_sousnut17 = df_sousnut['Valeur'][df_sousnut['Année'] == '2016-2018'].sum()
pop_sousnut17 = pop_sousnut17 *1e6
pop_sousnut17

873300000.0

*873,30 Millions de personnes dans le monde sont en sous-nutrition sur la période 2017*

#### 10.3 Proportion de la population mondiale étant en sous-nutrition en 2013

In [48]:
proportion_pop_sousnut2013 = round(pop_sousnut13 / population_mondiale * 100, 2)
proportion_pop_sousnut2013

12.6

*Soit 12,6% de la population mondiale en 2013*

#### 10.4 Proportion de la population mondiale étant en sous-nutrition en 2017

In [49]:
proportion_pop_sousnut2017 = round(pop_sousnut17 / population_mondiale * 100, 2)
proportion_pop_sousnut2017

12.48

*Soit 12,48% de la population mondiale en 2017*

#### 10.5 Liste des produits (ainsi que leur code) considérés comme des céréales selon la FAO

Les Céréales seront identifiées à l'aide d'une colonne de type booléenne nommée "is_cereal".

In [50]:
# Création d'un dataframe (cereal )à partir des données extraites de la FAO
cereal = pd.read_csv('p3_data/cereal_item_list.csv')
cereal

Code Domaine              Domaine  Code Pays   Pays  Code Élément  \
0  FBS          Bilans Alimentaires  5000       Monde  5511           
1  FBS          Bilans Alimentaires  5000       Monde  5511           
2  FBS          Bilans Alimentaires  5000       Monde  5511           
3  FBS          Bilans Alimentaires  5000       Monde  5511           
4  FBS          Bilans Alimentaires  5000       Monde  5511           
5  FBS          Bilans Alimentaires  5000       Monde  5511           
6  FBS          Bilans Alimentaires  5000       Monde  5511           
7  FBS          Bilans Alimentaires  5000       Monde  5511           
8  FBS          Bilans Alimentaires  5000       Monde  5511           

      Élément  Code Produit           Produit  Code Année  Année  \
0  Production  2511          Blé               2013        2013    
1  Production  2805          Riz (Eq Blanchi)  2013        2013    
2  Production  2513          Orge              2013        2013    
3  Production  2514          Maïs              2013        2013    
4  Production  2515          Seigle            2013        2013    
5  Production  2516          Avoine            2013        2013    
6  Production  2517          Millet            2013        2013    
7  Production  2518          Sorgho            2013        2013    
8  Production  2520          Céréales, Autres  2013        2013    

                Unité   Valeur Symbole  \
0  Milliers de tonnes  708443   A        
1  Milliers de tonnes  495447   A        
2  Milliers de tonnes  142460   A        
3  Milliers de tonnes  1017031  A        
4  Milliers de tonnes  16711    A        
5  Milliers de tonnes  23813    A        
6  Milliers de tonnes  28909    A        
7  Milliers de tonnes  62118    A        
8  Milliers de tonnes  28415    A        

                                                                   Description du Symbole  
0  Agrégat, peut inclure des données officielles, semi-officielles, estimées ou calculées  
1  Agrégat, peut inclure des données officielles, semi-officielles, estimées ou calculées  
2  Agrégat, peut inclure des données officielles, semi-officielles, estimées ou calculées  
3  Agrégat, peut inclure des données officielles, semi-officielles, estimées ou calculées  
4  Agrégat, peut inclure des données officielles, semi-officielles, estimées ou calculées  
5  Agrégat, peut inclure des données officielles, semi-officielles, estimées ou calculées  
6  Agrégat, peut inclure des données officielles, semi-officielles, estimées ou calculées  
7  Agrégat, peut inclure des données officielles, semi-officielles, estimées ou calculées  
8  Agrégat, peut inclure des données officielles, semi-officielles, estimées ou calculées

In [51]:
# Somme des valeurs de production de céréales sur l'année 2013 exprimée en Milliers de tonnes
cereal['Valeur'].sum()

2523347

In [52]:
print("Soit une production de céréales de 2,5 Milliards de tonnes en 2013")

Soit une production de céréales de 2,5 Milliards de tonnes en 2013


In [53]:
# Création de la liste des Céréales 
cereal_item_list = cereal['Produit'].tolist()
cereal_item_list

['Blé',
 'Riz (Eq Blanchi)',
 'Orge',
 'Maïs',
 'Seigle',
 'Avoine',
 'Millet',
 'Sorgho',
 'Céréales, Autres']

#### 10.6 Création d'une colonne "is_cereal" dans le dataframe (df1) des disponibilités alimentaires

In [54]:
# Création d'une liste vide nommée is_cereal
# Boucle itérative pour faire matcher la liste des Céréales (cereal_item_list) aux produits du dataframe
is_cereal = []
for row in df1['Produit'] :
    if row in cereal_item_list :
        is_cereal.append(True)
    else : 
        is_cereal.append(False)

In [55]:
df1['is_cereal'] = is_cereal

In [56]:
df1.head(10)

Élément  Code Pays     Pays  Code Produit           Produit   Origine  \
0        1          Arménie  2511          Blé               Végétale   
1        1          Arménie  2513          Orge              Végétale   
2        1          Arménie  2514          Maïs              Végétale   
3        1          Arménie  2515          Seigle            Végétale   
4        1          Arménie  2516          Avoine            Végétale   
5        1          Arménie  2517          Millet            Végétale   
6        1          Arménie  2518          Sorgho            Végétale   
7        1          Arménie  2520          Céréales, Autres  Végétale   
8        1          Arménie  2531          Pommes de Terre   Végétale   
9        1          Arménie  2532          Manioc            Végétale   

Élément  Population  Année  Aliments pour animaux  Autres Utilisations  \
0        2977        2013   93.0                   0.0                   
1        2977        2013   137.0                  26.0                  
2        2977        2013   96.0                  NaN                    
3        2977        2013   1.0                   NaN                    
4        2977        2013   4.0                   NaN                    
5        2977        2013   0.0                   NaN                    
6        2977        2013   0.0                   NaN                    
7        2977        2013   1.0                    12.0                  
8        2977        2013   225.0                  16.0                  
9        2977        2013   0.0                   NaN                    

Élément  Disponibilité alimentaire (Kcal/personne/jour)  ...  Nourriture  \
0        1024.0                                          ...  389.0        
1        0.0                                             ...  0.0          
2        0.0                                             ...  0.0          
3        1.0                                             ...  0.0          
4        2.0                                             ...  1.0          
5       NaN                                              ... NaN           
6       NaN                                              ... NaN           
7        14.0                                            ...  6.0          
8        82.0                                            ...  134.0        
9       NaN                                              ... NaN           

Élément  Pertes  Production  Semences  Traitement  Variation de stock  \
0        32.0    312.0       30.0      10.0       -118.0                
1        15.0    189.0       14.0      7.0         0.0                  
2        7.0     21.0        0.0      NaN         NaN                   
3        0.0     1.0         0.0      NaN          0.0                  
4        0.0     5.0         0.0      NaN         NaN                   
5       NaN     NaN         NaN       NaN         NaN                   
6       NaN     NaN         NaN       NaN         NaN                   
7        1.0     17.0        1.0      NaN          0.0                  
8        96.0    661.0       123.0     78.0        15.0                 
9       NaN     NaN         NaN       NaN         NaN                   

Élément  Disponibilité alimentaire (Kcal/an)  \
0        1.112684e+09                          
1        0.000000e+00                          
2        0.000000e+00                          
3        1.086605e+06                          
4        2.173210e+06                          
5       NaN                                    
6       NaN                                    
7        1.521247e+07                          
8        8.910161e+07                          
9       NaN                                    

Élément  Disponibilité de protéines en quantité (Kg/an)  \
0        33163.18460                                      
1        0.00000                                          
2        10.86605                 

### 11 : Quelle proportion (en termes de poids) est destinée à l'alimentation animale ?
En ne prenant en compte que les céréales destinées à l'alimentation (humaine et animale). 

In [57]:
# Création du dataframe (df100) selon la restriction des produits céréaliers 
df100 = df1[df1.is_cereal == True]

#### 11.1 Part des céréales destinée à l'alimentation animale sur l'année 2013

In [58]:
cer_animal2013 = df100['Aliments pour animaux'].sum() 
cer_animal2013 #valeur exprimée en Milliers de tonnes / an

873535.0

#### 11.2 Part des céréales destinée à l'alimentation humaine sur l'année 2013

In [59]:
cer_humain2013 = df100['Disponibilité intérieure'].sum() 
cer_humain2013 #valeur exprimée en Milliers de tonnes / an

2406999.0

#### 11.3 Proportion de la disponibilité mondiale des céréales destinée à l'alimentation animale sur l'année 2013

In [60]:
round(873535.0 / 2406999.0 * 100, 2)

36.29

*36% de la disponibilité mondiale des céréales est destinée à l'alimentation animale.*

#### 11.4 Sélection des données bilans alimentaires concernant uniquement les pays qui recensent des personnes en sous-nutrition (source FAO). 

In [61]:
# Création d'un dataframe complet selon les conditions restrictives des personnes en sous-nutrition    
# Réduction du dataframe relatif à la sous-nutrition (df_sousnut) sur l'année 2013
df111 = df_sousnut[df_sousnut['Code année'] == 20122014] 
df111.head()

Code Domaine                             Domaine  Code zone  \
0   FS           Données de la sécurité alimentaire  2           
5   FS           Données de la sécurité alimentaire  202         
10  FS           Données de la sécurité alimentaire  3           
15  FS           Données de la sécurité alimentaire  4           
20  FS           Données de la sécurité alimentaire  79          

              Zone  Code Élément Élément  Code Produit  \
0   Afghanistan     6132          Valeur  210011         
5   Afrique du Sud  6132          Valeur  210011         
10  Albanie         6132          Valeur  210011         
15  Algérie         6132          Valeur  210011         
20  Allemagne       6132          Valeur  210011         

                                                               Produit  \
0   Nombre de personnes sous-alimentées (millions) (moyenne sur 3 ans)   
5   Nombre de personnes sous-alimentées (millions) (moyenne sur 3 ans)   
10  Nombre de personnes sous-alimentées (millions) (moyenne sur 3 ans)   
15  Nombre de personnes sous-alimentées (millions) (moyenne sur 3 ans)   
20  Nombre de personnes sous-alimentées (millions) (moyenne sur 3 ans)   

    Code année      Année     Unité  Valeur Symbole Description du Symbole  \
0   20122014    2012-2014  millions  7.9     F       Estimation FAO          
5   20122014    2012-2014  millions  2.6     F       Estimation FAO          
10  20122014    2012-2014  millions  0.2     F       Estimation FAO          
15  20122014    2012-2014  millions  1.7     F       Estimation FAO          
20  20122014    2012-2014  millions NaN      NR      Non rapportée           

    Note  
0  NaN    
5  NaN    
10 NaN    
15 NaN    
20 NaN

In [62]:
# Suppression des NaN sur la colonne "Valeur" pour ne comptabiliser les pays non sous-alimentés 
# Comme l'exemple ci-dessus de l'Allemagne
df111 = df111.dropna(subset=['Valeur'])

In [63]:
# Identification des Pays dans lesquels la FAO recense des personnes en sous-nutrition
# 120 Pays sont concernés
print(df111['Code zone'].nunique())
list_pays_sousnut = df111['Code zone'].tolist()

120


In [64]:
# Création d'une colonne pour identifier dans nos données des bilans alimentaires les personnes étant en sous-nutrition
# Création d'une liste vide nommée pays_sousnut
# Boucle itérative pour faire matcher la liste des 120 pays (list_pays_sousnut) aux données des bilans alimentaires
pays_sousnut = []

for code in df1['Code Pays'] :
    if code in list_pays_sousnut :
        pays_sousnut.append(True)
    else : 
        pays_sousnut.append(False)

In [65]:
# Création d'un dataframe des bilans alimentaires avec mention des populations étant en sous-nutrition
# Réduction selon la condition de sous-nutrition des populations concernées
df112 = df1
df112['sous_nut'] = pays_sousnut
df112[df112['sous_nut'] == True]
df112.head()

Élément  Code Pays     Pays  Code Produit Produit   Origine  Population  \
0        1          Arménie  2511          Blé     Végétale  2977         
1        1          Arménie  2513          Orge    Végétale  2977         
2        1          Arménie  2514          Maïs    Végétale  2977         
3        1          Arménie  2515          Seigle  Végétale  2977         
4        1          Arménie  2516          Avoine  Végétale  2977         

Élément  Année  Aliments pour animaux  Autres Utilisations  \
0        2013   93.0                   0.0                   
1        2013   137.0                  26.0                  
2        2013   96.0                  NaN                    
3        2013   1.0                   NaN                    
4        2013   4.0                   NaN                    

Élément  Disponibilité alimentaire (Kcal/personne/jour)  ...  Pertes  \
0        1024.0                                          ...  32.0     
1        0.0                                             ...  15.0     
2        0.0                                             ...  7.0      
3        1.0                                             ...  0.0      
4        2.0                                             ...  0.0      

Élément  Production  Semences  Traitement  Variation de stock  \
0        312.0       30.0      10.0       -118.0                
1        189.0       14.0      7.0         0.0                  
2        21.0        0.0      NaN         NaN                   
3        1.0         0.0      NaN          0.0                  
4        5.0         0.0      NaN         NaN                   

Élément  Disponibilité alimentaire (Kcal/an)  \
0        1.112684e+09                          
1        0.000000e+00                          
2        0.000000e+00                          
3        1.086605e+06                          
4        2.173210e+06                          

Élément  Disponibilité de protéines en quantité (Kg/an)  \
0        33163.18460                                      
1        0.00000                                          
2        10.86605                                         
3        21.73210                                         
4        97.79445                                         

Élément  Disponibilité alimentaire en quantité (Kg/an)  is_cereal  sous_nut  
0        388796.20                                      True       True      
1        0.00                                           True       True      
2        89.31                                          True       True      
3        357.24                                         True       True      
4        1101.49                                        True       True      

[5 rows x 27 columns]

#### 11.5 Liste des 15 produits les plus exportés par ce groupe de pays sur l'année 2013

In [66]:
# Agrégation des données sur les produits pour tous les pays confondus
# Restriction sur les données d'exportation de ces produits
# Projection sur les données recherchées…
df112 = df112.groupby('Produit').sum().reset_index()
df112 = df112.sort_values(by=['Exportations - Quantité'], ascending=False)

df112 = df112[['Code Produit', 'Produit', 'Exportations - Quantité']].reset_index()
df112[:15]

Élément  index  Code Produit              Produit  Exportations - Quantité
0        12     436914        Blé                  198380.0               
1        52     437436        Maïs                 128787.0               
2        48     495552        Lait - Excl Beurre   125098.0               
3        81     424130        Soja                 106912.0               
4        83     442308        Sucre Eq Brut        68320.0                
5        39     438090        Huile de Palme       50673.0                
6        49     453270        Légumes, Autres      44535.0                
7        62     434749        Orge                 41585.0                
8        79     488070        Riz (Eq Blanchi)     41445.0                
9        61     451703        Oranges, Mandarines  37489.0                
10       51     422844        Manioc               36052.0                
11       25     456750        Fruits, Autres       30165.0                
12       76     440394        Pommes de Terre      29259.0                
13       73     477999        Poissons Pelagiques  23134.0                
14       89     449973        Tomates              21987.0

#### 11.6 Sélection des 200 plus grandes importations de ces produits sur l'année 2013
La sélection sera établie sur les données des bilans alimentaires au niveau mondial (dataframe df1)

(1 importation = une quantité d'un produit donné importée par un pays donné sur l'année choisie)

In [67]:
# Création du dataframe (df113) basé sur les bilans alimentaires au niveau mondial
df113 = df1

In [68]:
# Sélection des 15 produits précédemment trouvés sur l'année 2013
# (15 produits les plus exportés par les Pays dont la Population est en sous-nutrition)
list_prod_import = df112['Produit'][:15].tolist()
list_prod_import

['Blé',
 'Maïs',
 'Lait - Excl Beurre',
 'Soja',
 'Sucre Eq Brut',
 'Huile de Palme',
 'Légumes, Autres',
 'Orge',
 'Riz (Eq Blanchi)',
 'Oranges, Mandarines',
 'Manioc',
 'Fruits, Autres',
 'Pommes de Terre',
 'Poissons Pelagiques',
 'Tomates']

In [69]:
# Création d'une colonne pour identifier dans nos données des bilans alimentaires les 15 produits
# Création d'une liste vide nommée prod_import_sousnut
# Boucle itérative pour faire matcher la liste des 15 produits (list_prod_import) aux données des bilans alimentaires

prod_import_sousnut = []
for i in df1['Produit'] :
    if i in list_prod_import :
        prod_import_sousnut.append(True)
    else :
        prod_import_sousnut.append(False)

In [70]:
# Ajout de la colonne booléenne prod_import_sousnut
df113['prod_import_sousnut'] = prod_import_sousnut

In [71]:
# Réduction sur la condition de présence des 15 produits recherchés
df113 = df113[df113['prod_import_sousnut'] == True]

In [72]:
# Projection sur les catégories Pays, Produit et Importations
# Mais pour la suite de l'étude nous devons conserver un df avec une projection plus large...
df113 = df113[['Pays', 'Produit', 'Importations - Quantité', 'Autres Utilisations', 'Disponibilité intérieure', 'Aliments pour animaux', 'Nourriture']]

In [73]:
# Filtre sur les quantités importées
# Sélection des 200 plus grandes importations sur ces produits
df113 = df113.sort_values(by='Importations - Quantité', ascending=False)[:200]
df1130 = df113[['Pays', 'Produit', 'Importations - Quantité']].reset_index()
df1130

Élément  index                 Pays           Produit  Importations - Quantité
0        2509   Chine, continentale  Soja              63381.0                
1        2496   Chine, continentale  Manioc            29046.0                
2        6965   Japon                Maïs              14403.0                
3        12012  Arabie saoudite      Orge              10547.0                
4        3674   Égypte               Blé               10331.0                
..        ...      ...               ...                   ...                
195      12195  Sénégal              Riz (Eq Blanchi)  1120.0                 
196      7509   République de Corée  Soja              1118.0                 
197      7534   République de Corée  Légumes, Autres   1105.0                 
198      6137   Indonésie            Manioc            1102.0                 
199      7240   Kenya                Blé               1092.0                 

[200 rows x 4 columns]

#### 11.7 Sélection de ces importations par produit, afin d'avoir une table contenant 1 ligne pour chacun des 15 produits. 

In [74]:
# Création d'un nouveau dataframe pour conserver cette sélection
# Projection sur les deux colonnes recherchées
df114 = df113.groupby('Produit').sum().reset_index()
df1140 = df114[['Produit', 'Importations - Quantité']]
df1140

Élément              Produit  Importations - Quantité
0        Blé                  146539.0               
1        Fruits, Autres       15714.0                
2        Huile de Palme       29747.0                
3        Lait - Excl Beurre   87377.0                
4        Légumes, Autres      27116.0                
5        Manioc               34655.0                
6        Maïs                 101416.0               
7        Oranges, Mandarines  22945.0                
8        Orge                 22387.0                
9        Poissons Pelagiques  13445.0                
10       Pommes de Terre      16895.0                
11       Riz (Eq Blanchi)     12911.0                
12       Soja                 93455.0                
13       Sucre Eq Brut        34208.0                
14       Tomates              7143.0

#### 11.8 Ratio entre la quantité destinée aux "Autres utilisations" (Other uses) et la disponibilité intérieure.

In [75]:
# Création d'un dataframe rajoutant une colonne 'Ratio Autres Util/Dispo int'
df115 = df114
df115['Ratio Autres Util/Dispo int'] = df115['Autres Utilisations'] / df115['Disponibilité intérieure']
df1150 = df115[['Produit', 'Importations - Quantité', 'Ratio Autres Util/Dispo int']]
df1150 = df1150.sort_values(by='Ratio Autres Util/Dispo int', ascending=False)
df1150

Élément              Produit  Importations - Quantité  \
2        Huile de Palme       29747.0                   
5        Manioc               34655.0                   
6        Maïs                 101416.0                  
10       Pommes de Terre      16895.0                   
13       Sucre Eq Brut        34208.0                   
3        Lait - Excl Beurre   87377.0                   
0        Blé                  146539.0                  
11       Riz (Eq Blanchi)     12911.0                   
12       Soja                 93455.0                   
8        Orge                 22387.0                   
7        Oranges, Mandarines  22945.0                   
1        Fruits, Autres       15714.0                   
4        Légumes, Autres      27116.0                   
9        Poissons Pelagiques  13445.0                   
14       Tomates              7143.0                    

Élément  Ratio Autres Util/Dispo int  
2        0.724803                     
5        0.302335                     
6        0.264426                     
10       0.065253                     
13       0.038188                     
3        0.036166                     
0        0.032140                     
11       0.017190                     
12       0.003284                     
8        0.000407                     
7        0.000051                     
1        0.000000                     
4        0.000000                     
9        0.000000                     
14       0.000000

#### 11.9 Ratio entre la quantité destinée à la nourriture animale et la quantité destinée à la nourriture (animale + humaine)

In [76]:
# Création d'un dataframe rajoutant une colonne 'Ratio Autres Util/Dispo int'
df116 = df114
df116['Ratio Nour Ani/Nour globale'] = df116['Aliments pour animaux'] / df116['Nourriture']
df1160 = df116[['Produit', 'Importations - Quantité', 'Ratio Nour Ani/Nour globale']]
df1160 = df1160.sort_values(by='Ratio Nour Ani/Nour globale', ascending=False)
df1160

Élément              Produit  Importations - Quantité  \
8        Orge                 22387.0                   
6        Maïs                 101416.0                  
9        Poissons Pelagiques  13445.0                   
12       Soja                 93455.0                   
5        Manioc               34655.0                   
0        Blé                  146539.0                  
11       Riz (Eq Blanchi)     12911.0                   
3        Lait - Excl Beurre   87377.0                   
10       Pommes de Terre      16895.0                   
4        Légumes, Autres      27116.0                   
1        Fruits, Autres       15714.0                   
2        Huile de Palme       29747.0                   
7        Oranges, Mandarines  22945.0                   
13       Sucre Eq Brut        34208.0                   
14       Tomates              7143.0                    

Élément  Ratio Nour Ani/Nour globale  
8        59.624260                    
6        7.097826                     
9        3.087630                     
12       1.825566                     
5        1.648358                     
0        0.321980                     
11       0.104570                     
3        0.091055                     
10       0.050057                     
4        0.049091                     
1        0.000000                     
2        0.000000                     
7        0.000000                     
13       0.000000                     
14       0.000000

### 12. Identification des 3 produits qui ont la plus grande valeur pour chacun des 2 ratios 

#### 12.1 Les 3 produits dont le ratio Autres utilisations / Disponibilité intérieure ont la plus grande valeur

In [77]:
# Utilisation du dataframe df1150
df1150.head(3)

Élément         Produit  Importations - Quantité  Ratio Autres Util/Dispo int
2        Huile de Palme  29747.0                  0.724803                   
5        Manioc          34655.0                  0.302335                   
6        Maïs            101416.0                 0.264426

#### 12.2 Les 3 produits dont le ratio Nourriture animale / Nourriture (animale + humaine) ont la plus grande valeur

In [78]:
# Utilisation du dataframe df1160
df1160.head(3)

Élément              Produit  Importations - Quantité  \
8        Orge                 22387.0                   
6        Maïs                 101416.0                  
9        Poissons Pelagiques  13445.0                   

Élément  Ratio Nour Ani/Nour globale  
8        59.624260                    
6        7.097826                     
9        3.087630

### 13. Combien de tonnes de céréales pourraient être libérées si les USA diminuaient leur production de produits animaux de 10% ?

In [79]:
# Création d'un dataframe à partir du df100
# Double restriction sur la présence de Céréales et le pays USA
df1000 = df100
df1000 = df1000[(df1000['is_cereal'] == True) & (df1000["Pays"] == "États-Unis d'Amérique")]
df1000

Élément  Code Pays                   Pays  Code Produit           Produit  \
14263    231        États-Unis d'Amérique  2511          Blé                
14264    231        États-Unis d'Amérique  2513          Orge               
14265    231        États-Unis d'Amérique  2514          Maïs               
14266    231        États-Unis d'Amérique  2515          Seigle             
14267    231        États-Unis d'Amérique  2516          Avoine             
14268    231        États-Unis d'Amérique  2517          Millet             
14269    231        États-Unis d'Amérique  2518          Sorgho             
14270    231        États-Unis d'Amérique  2520          Céréales, Autres   
14355    231        États-Unis d'Amérique  2805          Riz (Eq Blanchi)   

Élément   Origine  Population  Année  Aliments pour animaux  \
14263    Végétale  320051      2013   6196.0                  
14264    Végétale  320051      2013   1440.0                  
14265    Végétale  320051      2013   128024.0                
14266    Végétale  320051      2013   184.0                   
14267    Végétale  320051      2013   1430.0                  
14268    Végétale  320051      2013   394.0                   
14269    Végétale  320051      2013   2351.0                  
14270    Végétale  320051      2013   77.0                    
14355    Végétale  320051      2013  NaN                      

Élément  Autres Utilisations  Disponibilité alimentaire (Kcal/personne/jour)  \
14263    16.0                 599.0                                            
14264    0.0                  5.0                                              
14265    137023.0             92.0                                             
14266    0.0                  2.0                                              
14267   NaN                   19.0                                             
14268   NaN                  NaN                                               
14269   NaN                   7.0                                              
14270   NaN                   5.0                                              
14355    142.0                72.0                                             

Élément  ...  Nourriture  Pertes  Production  Semences  Traitement  \
14263    ...  25742.0    NaN      57967.0     2096.0    0.0          
14264    ...  160.0      NaN      4683.0      107.0     3161.0       
14265    ...  3917.0     NaN      353699.0    582.0     23230.0      
14266    ...  87.0       NaN      195.0       76.0     NaN           
14267    ...  1265.0     NaN      1016.0      103.0    NaN           
14268    ... NaN          22.0    418.0      NaN       NaN           
14269    ...  236.0      NaN      9882.0      22.0      1753.0       
14270    ...  200.0      NaN      81.0        3.0      NaN           
14355    ...  2203.0      200.0   5745.0      109.0     464.0        

Élément  Variation de stock  Disponibilité alimentaire (Kcal/an)  \
14263    5284.0              6.997435e+10                          
14264    16.0                5.840931e+08                          
14265   -39863.0             1.074731e+10                          
14266   -76.0                2.336372e+08                          
14267    32.0                2.219554e+09                          
14268   NaN                 NaN                                    
14269   -3220.0              8.177303e+08                          
14270    0.0                 5.840931e+08                          
14355   -27.0                8.410940e+09                          

Élément  Disponibilité de protéines en quantité (Kg/an)  \
14263    2.260440e+06                                     
14264    1.635461e+04                                     
14265    1.950871e+05                                     
14266    8.177303e+03                                     
14267    9.228671e+04                                     
14268   NaN                                               
14

In [80]:
# Estimation du total de céréales destiné à l'alimentation animale
# Conversion des valeurs en tonnes (au lieu de milliers de tonnes)
prod_cer = (df1000['Aliments pour animaux'] * 1000000).sum()

# 10% de la production
prod_cer * .10

14009600000.0

*Plus de 14 Milliards de tonnes de Céréales pourraient être disponibles*

### 14. Proportion du Manioc en Thaïlande exportée dans le Monde et proportion de personnes en sous-nutrition

#### 14.1 Proportion du Manioc en Thaïlande exportée dans le Monde

In [81]:
# Création d'un dataframe (df_thai_manioc) à partir des bilans alimentaires du dataframe df1
# Restriction suivant le Pays Thaïlande et le produit Manioc
df_thai_manioc = df1[(df1['Pays'] == 'Thaïlande') & (df1['Produit'] == 'Manioc')]
df_thai_manioc = df_thai_manioc[['Pays', 'Produit', 'Exportations - Quantité']]
df_thai_manioc

Élément       Pays Produit  Exportations - Quantité
13344    Thaïlande  Manioc  25214.0

In [82]:
# Les valeurs "Exportations - Quantité" sont exprimées en Milliers de tonnes
# Conversion en tonnes
export_manioc_thai = 25214 * 1000000
export_manioc_thai

25214000000

*25 Milliards de tonnes de Manioc exportées par la Thaïlande*

#### 14.2 Proportion des exportations du Manioc par la Thaïlande / aux exportations mondiales du Manioc

In [83]:
# Création d'une variable propre au cumul des exportations de Manioc dans le Monde
export_manioc = df1['Exportations - Quantité'][df1['Produit'] == 'Manioc'].sum()
export_manioc = export_manioc * 1000000
export_manioc

36052000000.0

*36 Milliards de tonnes de Manioc exportées dans le monde*

*>>> En conclusion plus de 69% du Manioc exportés dans le monde provient de Thaïlande*

#### 14.3 Pays exportateurs de Manioc dans le monde sur l'année 2013

In [84]:
# Création d'un dataframe (df_pays_manioc) pour identifier et quantifier ces pays
df_pays_manioc = (df1[(df1['Produit'] == 'Manioc') & (df1['Exportations - Quantité'] != 0)])
df_pays_manioc = df_pays_manioc.dropna(subset=['Exportations - Quantité'])
df_pays_manioc = df_pays_manioc[['Pays', 'Exportations - Quantité']].sort_values(by='Exportations - Quantité', ascending=False)
df_pays_manioc

Élément                         Pays  Exportations - Quantité
13344    Thaïlande                    25214.0                
14724    Viet Nam                     8973.0                 
6137     Indonésie                    820.0                  
7345     Cambodge                     254.0                  
10719    Paraguay                     164.0                  
2777     Costa Rica                   123.0                  
13154    Chine, Taiwan Province de    82.0                   
5669     Chine - RAS de Hong-Kong     80.0                   
13991    Ouganda                      47.0                   
2496     Chine, continentale          41.0                   
1342     Brésil                       38.0                   
9613     Pays-Bas                     38.0                   
6040     Inde                         21.0                   
10065    Nicaragua                    16.0                   
4685     Allemagne                    14.0                   
2235     Sri Lanka                    12.0                   
10251    Nigéria                      11.0                   
4871     Ghana                        9.0                    
14272    États-Unis d'Amérique        9.0                    
13248    République-Unie de Tanzanie  9.0                    
3588     Équateur                     8.0                    
10811    Pérou                        7.0                    
3966     Fidji                        7.0                    
15161    Belgique                     6.0                    
8974     Mexique                      5.0                    
14086    Royaume-Uni                  5.0                    
10906    Philippines                  4.0                    
11095    Portugal                     4.0                    
11509    Rwanda                       4.0                    
8427     Malaisie                     4.0                    
6596     Italie                       3.0                    
3683     Égypte                       3.0                    
4153     France                       2.0                    
2592     Colombie                     2.0                    
3149     Bénin                        2.0                    
5574     Honduras                     2.0                    
12476    Afrique du Sud               2.0                    
12571    Espagne                      1.0                    
13899    Émirats arabes unis          1.0                    
5765     Hongrie                      1.0                    
10530    Panama                       1.0                    
1878     Cameroun                     1.0                    
15068    Zambie                       1.0                    
1605     Brunéi Darussalam            1.0

In [85]:
df_pays_manioc['Pays'].nunique()

44

*44 Pays exportent le Manioc* 

*Les 3 pays qui exportent le plus le Manioc sont la Thaïlande, le Viet Nam et l'Indonésie.*

#### 14.4 Proportion des personnes en Thaïlande rencensées comme étant en sous-nutrition

In [86]:
# Années 2012/2013/2014 Nombre de personnes sous-alimentées (millions) (moyenne sur 3 ans)

In [87]:
# Projection et restriction à partir du dataframe (df_sousnut) extrait des données de sécurité alimentaire de la FAO
df_sousnut[['Zone', 'Unité', 'Valeur']][(df_sousnut['Zone'] == 'Thaïlande') & (df_sousnut['Code année'] == 20122014)]

Zone     Unité  Valeur
930  Thaïlande  millions  5.6

*5,6 Millions de personnes en Thaïlande recensées en état de sous-nutrition entre 2012 et 2014*

In [88]:
# Années 2016/2017/2018 Nombre de personnes sous-alimentées (millions) (moyenne sur 3 ans)

In [89]:
# Projection et restriction à partir du dataframe (df_sousnut) extrait des données de sécurité alimentaire de la FAO
df_sousnut[['Zone', 'Unité', 'Valeur']][(df_sousnut['Zone'] == 'Thaïlande') & (df_sousnut['Code année'] == 20162018)]

Zone     Unité  Valeur
934  Thaïlande  millions  5.4

*5,4 Millions de personnes en Thaïlande recensées en état de sous-nutrition entre 2016 et 2018*

In [99]:
# La population de la Thaïlande en 2013 est estimée à 67 Millions de personnes (visible sur df1)

*8,3% de la population en Thaïlande sur 2013 est considérée comme étant en sous-nutrition*

---

[Retour vers la page notebook précédente](https://github.com/nalron/project_public_health_study/blob/french_version/p3_notebook01.ipynb)

[Ou suite du projet vers l'intégration des données dans une base de données relationnelle via le langage SQL](https://github.com/nalron/project_public_health_study/blob/french_version/p3_notebook03.ipynb)